rtcpeerconnection.go

```diff
@@ -77,26 +99,24 @@ func (r *RTCPeerConnection) CreateOffer() error {
 	}
 
 	r.tlscfg = dtls.NewTLSCfg()
-	r.iceUsername = util.RandSeq(16)
-	r.icePassword = util.RandSeq(32)
 
 	r.portsLock.Lock()
 	defer r.portsLock.Unlock()
 
 	candidates := []string{}
 	basePriority := uint16(rand.Uint32() & (1<<16 - 1))
-	for id, c := range ice.HostInterfaces() {
-		port, err := network.NewPort(c+":0", []byte(r.icePassword), r.tlscfg, r.generateChannel, r.iceStateChange)
-		if err != nil {
-			return err
+	for id, server := range r.config.ICEServers {
```

Comment:
I think r.ports will end up being `portsAndTURN`(bad name) but you still want to bind to the IP+Port that ice.HostInterfaces() return.

So for STUN you won't create any new connections (the hole punching will happen and you won't notice anything different, just be listening on local ports like if you were in the same LAN) but for TURN you will need dedicated connections.

Reply:
I see. I guess I'll need to add an identification if something is TURN or STUN; STUN servers are ignored effectivly ignored, as the local connections should suffice?